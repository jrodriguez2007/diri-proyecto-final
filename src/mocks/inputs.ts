// src/mocks/movements.ts

import { Logistic } from '../models/Logistic'
import { LogisticDetail } from '../models/LogisticDetail'

export const mocInputs: Logistic[] = [
  {
    id:           "00001",
    idDocumento:  "PE",
    idSerie:      "E001",
    numero:       "00001",
    fechaDocumento: "2025-03-06",
    idTipoMovimiento: "IC",
    tipoMovimiento:   "Ingreso por Compra",
    idOperador:       "+",
    idAlmacen:        1,
    almacen:          "ALM. PRINCIPAL",
    idResponsable:    6,
    documentoResponsable: "20678901234",
    responsable:      "J & M DISTRIBUIDORES",
    glosa:            "INGRESO DE PRUEBA",
    activo:           true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion:  null,
    detalle: [
      {
        idLinea:             1,
        idProducto:          1,
        codigo:              "PROD001",
        descripcionExtendida:"SAMSUNG ODYSSEY LS27CG552EUXEN 27\" LED QHD 165HZ FREESYNC CURVO",
        unidad:              "UNIDAD ",
        marca:               "SAMSUNG",
        modelo:              "ODYSSEY",
        cantidad:            2.0000,
      },
      {
        idLinea:             2,
        idProducto:          2,
        codigo:              "PROD002",
        descripcionExtendida:"SAMSUNG 49\" OLED - ODYSSEY OLED G9 G91SD S49DG910SU",
        unidad:              "UNIDAD ",
        marca:               "SAMSUNG",
        modelo:              "FREESYNC CURVO",
        cantidad:            1.0000,
      },
    ],
  },
  {
    id:           "00002",
    idDocumento:  "PE",
    idSerie:      "E001",
    numero:       "00002",
    fechaDocumento: "2025-03-16",
    idTipoMovimiento: "IC",
    tipoMovimiento:   "Ingreso por Compra",
    idOperador:       "+",
    idAlmacen:        1,
    almacen:          "ALM. PRINCIPAL",
    idResponsable:    7,
    documentoResponsable: "20789012345",
    responsable:      "DELTRON S.A.C.",
    glosa:            "INGRESO POR COMPRA",
    activo:           true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion:  null,
    detalle: [
      {
        idLinea:             1,
        idProducto:          1,
        codigo:              "PROD001",
        descripcionExtendida:"SAMSUNG ODYSSEY LS27CG552EUXEN 27'' LED QHD 165HZ FREESYNC CURVO",
        unidad:              "UNIDAD ",
        marca:               "SAMSUNG",
        modelo:              "ODYSSEY",
        cantidad:            3.0000,
      },
      {
        idLinea:             2,
        idProducto:          2,
        codigo:              "PROD002",
        descripcionExtendida:"SAMSUNG 49'' OLED - ODYSSEY OLED G9 G91SD S49DG910SU",
        unidad:              "UNIDAD ",
        marca:               "SAMSUNG",
        modelo:              "FREESYNC CURVO",
        cantidad:            5.0000,
      },
    ],
  },
  {
    id:           "00015",
    idDocumento:  "PE",
    idSerie:      "E001",
    numero:       "00015",
    fechaDocumento: "2025-03-24",
    idTipoMovimiento: "IC",
    tipoMovimiento:   "Ingreso por Compra",
    idOperador:       "+",
    idAlmacen:        1,
    almacen:          "ALM. PRINCIPAL",
    idResponsable:    4,
    documentoResponsable: "20456789012",
    responsable:      "ARC SERVICIO Y DISTRIBUIDORES S.A.",
    glosa:            "REGISTRO CON DETALLE",
    activo:           true,
    idUsuarioRegistro: "jperez",
    idUsuarioEdicion:  null,
    detalle: [
      {
        idLinea:             1,
        idProducto:          1,
        codigo:              "PROD001",
        descripcionExtendida:"SAMSUNG ODYSSEY LS27CG552EUXEN 27\" LED QHD 165HZ FREESYNC CURVO",
        unidad:              "UNIDAD ",
        marca:               "SAMSUNG",
        modelo:              "ODYSSEY",
        cantidad:            2.0000,
      },
      {
        idLinea:             2,
        idProducto:          3,
        codigo:              "PROD003",
        descripcionExtendida:"MONITOR LG ULTRAWIDE 34' CURVO, VA, WQHD, 100 HZ, 21:9",
        unidad:              "DOCENA",
        marca:               "SAMSUNG",
        modelo:              "FREESYNC CURVO",
        cantidad:            6.0000,
      },
      {
        idLinea:             3,
        idProducto:          1013,
        codigo:              "PROD005",
        descripcionExtendida:"MONITOR DE FANTASIA",
        unidad:              "CAJA",
        marca:               "LOGITECH",
        modelo:              "LIFT",
        cantidad:            4.0000,
      },
    ],
  },
]
