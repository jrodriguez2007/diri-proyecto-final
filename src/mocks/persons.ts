import { Person } from '../models/Person';

export const mockPersons: Person[] = [
  {
    id: 1,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20123456789",
    razonSocial: "SUMINISTROS INFORMÁTICOS S.A.C.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "carlos.garcia@example.com",
    telefono: "555-0101",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 2,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20234567890",
    razonSocial: "HI-FI IMPORTADORA E.I.R.L.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "ana.martinez@example.com",
    telefono: "555-0102",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 3,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20345678901",
    razonSocial: "DATA NETWORK S.A.C.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "luis.rodriguez@example.com",
    telefono: "555-0103",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 4,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20456789012",
    razonSocial: "ARC SERVICIO Y DISTRIBUIDORES S.A.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "maria.fernandez@example.com",
    telefono: "555-0104",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 5,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20567890123",
    razonSocial: "WEBMASTER Y SERVICIOS S.A.C.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "jorge.lopez@example.com",
    telefono: "555-0105",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 6,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20678901234",
    razonSocial: "J & M DISTRIBUIDORES",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "elena.ramirez@example.com",
    telefono: "555-0106",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 7,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20789012345",
    razonSocial: "DELTRON S.A.C.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "miguel.soto@example.com",
    telefono: "555-0107",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 8,
    nombres: "",
    apellidoPaterno: "",
    apellidoMaterno: "",
    numeroDocumento: "20890123456",
    razonSocial: "PC-LINK S.A.",
    idTipoPersona: "J",
    tipoPersona: "",
    correoElectronico: "laura.diaz@example.com",
    telefono: "555-0108",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 9,
    nombres: "Pedro",
    apellidoPaterno: "Castro",
    apellidoMaterno: "Ruiz",
    numeroDocumento: "46781159",
    razonSocial: "",
    idTipoPersona: "N",
    tipoPersona: "",
    correoElectronico: "pedro.castro@example.com",
    telefono: "555-0109",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 10,
    nombres: "Sofía",
    apellidoPaterno: "Ramirez",
    apellidoMaterno: "Flores",
    numeroDocumento: "70458930",
    razonSocial: "",
    idTipoPersona: "N",
    tipoPersona: "",
    correoElectronico: "sofia.ramirez@example.com",
    telefono: "555-0110",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 12,
    nombres: "Juan",
    apellidoPaterno: "Rodriguez",
    apellidoMaterno: "Mamani",
    numeroDocumento: "77005689",
    razonSocial: "",
    idTipoPersona: "N",
    tipoPersona: "",
    correoElectronico: "juan.rodriguez@example.com",
    telefono: "555-0110",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
  {
    id: 13,
    nombres: "Pedro",
    apellidoPaterno: "Nuñez",
    apellidoMaterno: "Sotomayor",
    numeroDocumento: "45789600",
    razonSocial: "",
    idTipoPersona: "N",
    tipoPersona: "",
    correoElectronico: "pedro.nunez@example.com",
    telefono: "555-0110",
    activo: true,
    idUsuarioRegistro: "jrodriguez",
    idUsuarioEdicion: "",
  },
];
