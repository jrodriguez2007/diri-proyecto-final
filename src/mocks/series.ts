import { Serie } from "../models/Serie";

export const mockSeries: Serie[] = [
  {
    id: "E001",
    idDocument: "PE",
    idSerie: "E001",
    number: 1,
    currentNumber: 100,
    sizeText: 5,
    status: true    
  },
  {
    id: "S001",
    idDocument: "PS",
    idSerie: "S001",
    number: 1,
    currentNumber: 200,
    sizeText: 5,
    status: true     
  },  
];