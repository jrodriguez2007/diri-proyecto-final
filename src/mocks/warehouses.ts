import { Warehouse } from '../models/Warehouse';


export const mockWarehouses: Warehouse[] = [
  {
    id: 1,
    nombre: 'ALM. PRINCIPAL',
  },
];
