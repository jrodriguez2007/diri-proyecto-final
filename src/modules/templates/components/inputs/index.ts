export * from './TreeTemplate';
export * from './SwitchTemplate';
export * from './ButtonTemplate';
export * from './HeaderButtonsTemplate';