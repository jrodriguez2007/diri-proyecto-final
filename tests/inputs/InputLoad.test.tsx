// tests/movements/InputLoad.test.tsx

import '@testing-library/jest-dom'
import React from 'react'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import { IntlProvider } from 'react-intl'
import esMessages from '../../src/lang/es.json'
import { afterEach, describe, expect, it, vi } from 'vitest'

import { mocInputs as mockInputs } from '../../src/mocks/inputs'

// 1) Mockeo de nuestro hook de traducción
vi.mock('../../src/hooks/useTransform', () => ({
  useTransform: () => ({ t: (id: string) => id })
}))

// 2) Mockeo de RTK-Query hooks de store de inputs (para evitar el “no redux context”)
vi.mock('../../src/modules/movements/store', () => ({
  useGetAllMovementInputsQuery: vi.fn(() => ({
    data: mockInputs,
    isLoading: false,
    error: undefined,
  })),
  useSearchInputsQuery: vi.fn(() => ({
    data: mockInputs,
    isLoading: false,
    error: undefined,
  })),
  // Además stubeamos estos para que useInputHandlers no falle
  useLazyGetInputByIdQuery: vi.fn(() => [() => ({ unwrap: async () => mockInputs[0] })]),
  useCreateInputMutation:        vi.fn(() => [() => ({ unwrap: async () => mockInputs[0] })]),
  useUpdateInputMutation:        vi.fn(() => [() => ({ unwrap: async () => mockInputs[0] })]),
  useDeleteInputMutation:        vi.fn(() => [() => ({ unwrap: async () => {}            })]),
  useUpdateStateInputMutation:   vi.fn(() => [() => ({ unwrap: async () => mockInputs[0] })]),
  useDeleteMovementInputMutation:vi.fn(() => [() => ({ unwrap: async () => {}            })]),
}))

// 3) Mockeo de hooks de negocio (useInputHandlers, columnas y búsqueda) para no invocar lógica pesada
vi.mock('../../src/modules/movements/hooks', () => ({
  useInputHandlers: () => ({
    handleCreate:      vi.fn(),
    handleEdit:        vi.fn(),
    handleFormSubmit:  vi.fn(),
    handleDelete:      vi.fn(),
    confirmDelete:     vi.fn(),
    handleUpdateState: vi.fn(),
    handleDeleteInput: vi.fn(),
  }),
  useInputColumns: () => [
    { title: 'Nº',         dataIndex: 'numero', key: 'numero' },
    { title: 'Fecha',      dataIndex: 'fechaDocumento', key: 'fechaDocumento' },
    { title: 'Movimiento', dataIndex: 'tipoMovimiento',  key: 'tipoMovimiento' },
    { title: 'Responsable',dataIndex: 'responsable',     key: 'responsable' },
  ],
  useInputSearch: vi.fn(() => ({
    data: mockInputs,
    isLoading: false,
    error: undefined,
  })),
}))

// 4) Mockeo de endpoints auxiliares (series, almacenes, proveedores, productos)
vi.mock('../../src/modules/series/store', () => ({
  useGetSeriesByIdDocumentQuery: () => ({ data: [], isLoading: false }),
}))
vi.mock('../../src/modules/warehouses/store', () => ({
  useGetAllWarehousesQuery: () => ({ data: [], isLoading: false }),
}))
vi.mock('../../src/modules/suppliers/store', () => ({
  useLazyGetPersonByDocumentQuery: vi.fn(() => [() => ({ unwrap: async () => ({ documentoResponsable: 'X' }) })]),
}))
vi.mock('../../src/modules/products/store', () => ({
  useLazyGetProductByCodeQuery: vi.fn(() => [() => ({ unwrap: async () => ({ codigo: 'P', cantidad: 1 }) })]),
}))

// 5) Importa **después** de todos los mocks
import { InputPage } from '../../src/modules/movements/pages/InputPage'

afterEach(() => {
  cleanup()
  vi.clearAllMocks()
})

describe('InputPage — Renderizado inicial', () => {


  it('Debe de mostrarse correctamente con valores por defecto', () => {
    const { container } = render(
      <IntlProvider locale="es" messages={esMessages}>
        <InputPage />
      </IntlProvider>
    )   
    expect( container ).toMatchSnapshot();
  }); 

  it('muestra el título y la tabla con los movimientos de ingreso', () => {
    render(
      <IntlProvider locale="es" messages={esMessages}>
        <InputPage />
      </IntlProvider>
    )

    // A) Título
    expect(screen.getByText(esMessages['app.movements.inputs.title'])).toBeInTheDocument()
    expect(screen.getByText(esMessages['buttons.new'])).toBeInTheDocument()
    // C) La tabla aparece y tiene tantas filas de cuerpo como movimientos
    const table = screen.getByRole('table')
    expect(table).toBeInTheDocument()

    const bodyRows = table.querySelectorAll('tbody tr')
    expect(bodyRows).toHaveLength(mockInputs.length)

    expect(screen.getByText('Nº')).toBeInTheDocument()
    expect(screen.getByText('Fecha')).toBeInTheDocument()
    expect(screen.getByText('Movimiento')).toBeInTheDocument()
    expect(screen.getByText('Responsable')).toBeInTheDocument()  

  })


})
