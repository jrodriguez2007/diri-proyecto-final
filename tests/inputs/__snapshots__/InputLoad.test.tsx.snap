// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`InputPage — Renderizado inicial > Debe de mostrarse correctamente con valores por defecto 1`] = `
<div>
  <div
    class="ant-layout css-dev-only-do-not-override-tjsggz"
    style="height: 100%; width: 100%; overflow: hidden;"
  >
    <header
      class="ant-layout-header css-dev-only-do-not-override-tjsggz"
      style="background: transparent; display: flex; justify-content: space-between; align-items: center; font-size: 24px; font-weight: bold; padding: 16px;"
    >
      <div>
        Movimientos de Ingreso
      </div>
      <div>
        <div
          class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-small ant-space-gap-col-small"
        >
          <div
            class="ant-space-item"
          >
            <span
              class="ant-input-affix-wrapper css-dev-only-do-not-override-tjsggz ant-input-outlined"
              style="width: 250px;"
            >
              <span
                class="ant-input-prefix"
              >
                <span
                  aria-label="search"
                  class="anticon anticon-search"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="search"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0011.6 0l43.6-43.5a8.2 8.2 0 000-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                    />
                  </svg>
                </span>
              </span>
              <input
                class="ant-input css-dev-only-do-not-override-tjsggz"
                placeholder="modules.form.placeholder.search"
                type="text"
                value=""
              />
              <span
                class="ant-input-suffix"
              >
                <button
                  class="ant-input-clear-icon ant-input-clear-icon-hidden"
                  tabindex="-1"
                  type="button"
                >
                  <span
                    aria-label="close-circle"
                    class="anticon anticon-close-circle"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="close-circle"
                      fill="currentColor"
                      fill-rule="evenodd"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M512 64c247.4 0 448 200.6 448 448S759.4 960 512 960 64 759.4 64 512 264.6 64 512 64zm127.98 274.82h-.04l-.08.06L512 466.75 384.14 338.88c-.04-.05-.06-.06-.08-.06a.12.12 0 00-.07 0c-.03 0-.05.01-.09.05l-45.02 45.02a.2.2 0 00-.05.09.12.12 0 000 .07v.02a.27.27 0 00.06.06L466.75 512 338.88 639.86c-.05.04-.06.06-.06.08a.12.12 0 000 .07c0 .03.01.05.05.09l45.02 45.02a.2.2 0 00.09.05.12.12 0 00.07 0c.02 0 .04-.01.08-.05L512 557.25l127.86 127.87c.04.04.06.05.08.05a.12.12 0 00.07 0c.03 0 .05-.01.09-.05l45.02-45.02a.2.2 0 00.05-.09.12.12 0 000-.07v-.02a.27.27 0 00-.05-.06L557.25 512l127.87-127.86c.04-.04.05-.06.05-.08a.12.12 0 000-.07c0-.03-.01-.05-.05-.09l-45.02-45.02a.2.2 0 00-.09-.05.12.12 0 00-.07 0z"
                      />
                    </svg>
                  </span>
                </button>
              </span>
            </span>
          </div>
          <div
            class="ant-space-item"
          >
            <button
              class="ant-btn css-dev-only-do-not-override-tjsggz ant-btn-primary ant-btn-color-primary ant-btn-variant-solid"
              type="button"
            >
              <span
                class="ant-btn-icon"
              >
                <span
                  aria-label="plus"
                  class="anticon anticon-plus"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="plus"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M482 152h60q8 0 8 8v704q0 8-8 8h-60q-8 0-8-8V160q0-8 8-8z"
                    />
                    <path
                      d="M192 474h672q8 0 8 8v60q0 8-8 8H160q-8 0-8-8v-60q0-8 8-8z"
                    />
                  </svg>
                </span>
              </span>
              Nuevo
            </button>
          </div>
        </div>
      </div>
    </header>
    <main
      class="ant-layout-content css-dev-only-do-not-override-tjsggz"
      style="display: flex; justify-content: center; align-items: flex-start; min-height: 100vh; padding: 20px; background: #fff; border-radius: 8px; box-shadow: 0 2px 8px rgba(0, 0, 0, 0.1); margin: 20px; overflow: hidden; box-sizing: border-box;"
    >
      <div
        style="width: 100%;"
      >
        <div
          class="ant-table-wrapper css-dev-only-do-not-override-tjsggz"
        >
          <div
            class="ant-spin-nested-loading css-dev-only-do-not-override-tjsggz"
          >
            <div
              class="ant-spin-container"
            >
              <div
                class="ant-table css-dev-only-do-not-override-tjsggz"
              >
                <div
                  class="ant-table-container"
                >
                  <div
                    class="ant-table-content"
                  >
                    <table
                      style="table-layout: auto;"
                    >
                      <colgroup>
                        <col
                          class="ant-table-expand-icon-col"
                        />
                      </colgroup>
                      <thead
                        class="ant-table-thead"
                      >
                        <tr>
                          <th
                            class="ant-table-cell ant-table-row-expand-icon-cell"
                          />
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Nº
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Fecha
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Movimiento
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            Responsable
                          </th>
                          <th
                            class="ant-table-cell"
                            scope="col"
                          >
                            table.header.actions
                          </th>
                        </tr>
                      </thead>
                      <tbody
                        class="ant-table-tbody"
                      >
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-gray-100"
                          data-row-key="00015"
                        >
                          <td
                            class="ant-table-cell ant-table-row-expand-icon-cell"
                          >
                            <button
                              aria-expanded="false"
                              aria-label="Expand row"
                              class="ant-table-row-expand-icon ant-table-row-expand-icon-collapsed"
                              type="button"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            00015
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            2025-03-24
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Ingreso por Compra
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            ARC SERVICIO Y DISTRIBUIDORES S.A.
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-white"
                          data-row-key="00002"
                        >
                          <td
                            class="ant-table-cell ant-table-row-expand-icon-cell"
                          >
                            <button
                              aria-expanded="false"
                              aria-label="Expand row"
                              class="ant-table-row-expand-icon ant-table-row-expand-icon-collapsed"
                              type="button"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            00002
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            2025-03-16
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Ingreso por Compra
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            DELTRON S.A.C.
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                        <tr
                          class="ant-table-row ant-table-row-level-0 bg-gray-100"
                          data-row-key="00001"
                        >
                          <td
                            class="ant-table-cell ant-table-row-expand-icon-cell"
                          >
                            <button
                              aria-expanded="false"
                              aria-label="Expand row"
                              class="ant-table-row-expand-icon ant-table-row-expand-icon-collapsed"
                              type="button"
                            />
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            00001
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            2025-03-06
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            Ingreso por Compra
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            J & M DISTRIBUIDORES
                          </td>
                          <td
                            class="ant-table-cell"
                          >
                            <div
                              class="ant-space css-dev-only-do-not-override-tjsggz ant-space-horizontal ant-space-align-center ant-space-gap-row-middle ant-space-gap-col-middle"
                            >
                              <div
                                class="ant-space-item"
                              >
                                <span
                                  aria-label="delete"
                                  class="anticon anticon-delete"
                                  role="img"
                                  tabindex="-1"
                                >
                                  <svg
                                    aria-hidden="true"
                                    data-icon="delete"
                                    fill="currentColor"
                                    focusable="false"
                                    height="1em"
                                    viewBox="64 64 896 896"
                                    width="1em"
                                  >
                                    <path
                                      d="M864 256H736v-80c0-35.3-28.7-64-64-64H352c-35.3 0-64 28.7-64 64v80H160c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h60.4l24.7 523c1.6 34.1 29.8 61 63.9 61h454c34.2 0 62.3-26.8 63.9-61l24.7-523H888c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zm-200 0H360v-72h304v72z"
                                    />
                                  </svg>
                                </span>
                              </div>
                            </div>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <ul
            class="ant-pagination ant-pagination-center css-dev-only-do-not-override-tjsggz"
          >
            <li
              aria-disabled="true"
              class="ant-pagination-prev ant-pagination-disabled"
              title="Previous Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="left"
                  class="anticon anticon-left"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="left"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
              tabindex="0"
              title="1"
            >
              <a
                rel="nofollow"
              >
                1
              </a>
            </li>
            <li
              aria-disabled="true"
              class="ant-pagination-next ant-pagination-disabled"
              title="Next Page"
            >
              <button
                class="ant-pagination-item-link"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <span
                  aria-label="right"
                  class="anticon anticon-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                    />
                  </svg>
                </span>
              </button>
            </li>
            <li
              class="ant-pagination-options"
            >
              <div
                aria-label="Page Size"
                class="ant-select ant-select-outlined ant-pagination-options-size-changer css-dev-only-do-not-override-tjsggz ant-select-single ant-select-show-arrow ant-select-show-search"
              >
                <div
                  class="ant-select-selector"
                >
                  <span
                    class="ant-select-selection-wrap"
                  >
                    <span
                      class="ant-select-selection-search"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-controls="rc_select_TEST_OR_SSR_list"
                        aria-expanded="false"
                        aria-haspopup="listbox"
                        aria-label="Page Size"
                        aria-owns="rc_select_TEST_OR_SSR_list"
                        autocomplete="off"
                        class="ant-select-selection-search-input"
                        id="rc_select_TEST_OR_SSR"
                        role="combobox"
                        type="search"
                        value=""
                      />
                    </span>
                    <span
                      class="ant-select-selection-item"
                      title="10 components.pagination.locale"
                    >
                      10 components.pagination.locale
                    </span>
                  </span>
                </div>
                <span
                  aria-hidden="true"
                  class="ant-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <span
                    aria-label="down"
                    class="anticon anticon-down ant-select-suffix"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="down"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M884 256h-75c-5.1 0-9.9 2.5-12.9 6.6L512 654.2 227.9 262.6c-3-4.1-7.8-6.6-12.9-6.6h-75c-6.5 0-10.3 7.4-6.5 12.7l352.6 486.1c12.8 17.6 39 17.6 51.7 0l352.6-486.1c3.9-5.3.1-12.7-6.4-12.7z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </main>
  </div>
</div>
`;
